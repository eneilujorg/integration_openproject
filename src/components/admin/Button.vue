<template>
	<button :disabled="isDisabled"
		:class="{ 'b-disabled': isDisabled }"
		v-on="$listeners">
		<div v-if="iconClass" class="b-icon" :class="iconClass" />
		<span>{{ t('integration_openproject', text) }}</span>
		<div v-if="isLoading" class="icon-loading" />
	</button>
</template>
<script>
export default {
	name: 'Button',
	props: {
		isLoading: {
			type: Boolean,
			default: false,
		},
		iconClass: {
			type: String,
			default: null,
		},
		text: {
			required: true,
			type: String,
		},
		isDisabled: {
			default: false,
			type: Boolean,
		},
	},
}
</script>
<style lang="scss" scoped>
button {
	display: flex;
	justify-content: center;
	align-items: center;
	font-size: .875rem;
	line-height: 1.25rem;
	font-weight: 400;
	position: relative;
	.b-icon {
		height: 16px;
		width: 16px;
		background-repeat: no-repeat;
		background-position: center;
		margin-right: 6px;
	}
	.icon-loading {
		min-height: 0;
	}
	.icon-loading:after {
		height: 14px;
		width: 14px;
		top: 6px;
		left: 20px;
	}
}

button.b-disabled {
	background: #CCCCCC !important;
	border: #CCCCCC !important;
	color: #FFFFFF !important;
}
</style>
