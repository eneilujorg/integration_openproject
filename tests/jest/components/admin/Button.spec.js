import { createLocalVue, mount } from '@vue/test-utils'
import Button from '../../../../src/components/admin/Button'

const localVue = createLocalVue()

describe('Admin Button', () => {
	it('should not show the loading icon or button icon if not provided', () => {
		const wrapper = getWrapper()
		expect(wrapper).toMatchSnapshot()
	})
	it('should show the loading icon when "isLoading" is set as "true"', () => {
		const wrapper = getWrapper({
			isLoading: true,
		})
		expect(wrapper).toMatchSnapshot()
	})
	it('should show the icon if the "iconClass" prop is provided with value', () => {
		const wrapper = getWrapper({
			iconClass: 'some class name',
		})
		expect(wrapper).toMatchSnapshot()
	})
	it('should show both button icon and loading icon if set as "true"', () => {
		const wrapper = getWrapper({
			iconClass: 'some class name',
			isLoading: true,
		})
		expect(wrapper).toMatchSnapshot()
	})
	it('should set the button as disabled when isDisabled prop is set as "true"', () => {
		const wrapper = getWrapper({
			isDisabled: true,
		})
		expect(wrapper).toMatchSnapshot()
	})
})

function getWrapper(props = {}) {
	return mount(Button, {
		localVue,
		propsData: {
			text: 'My Button',
			...props,
		},
		mocks: {
			t: (app, msg) => msg,
		},
	})
}
