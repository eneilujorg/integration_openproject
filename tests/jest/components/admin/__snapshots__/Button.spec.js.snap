// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Button should not show the loading icon or button icon if not provided 1`] = `
<button class="">
  <!----> <span>My Button</span>
  <!---->
</button>
`;

exports[`Admin Button should set the button as disabled when isDisabled prop is set as "true" 1`] = `
<button disabled="disabled" class="b-disabled">
  <!----> <span>My Button</span>
  <!---->
</button>
`;

exports[`Admin Button should show both button icon and loading icon if set as "true" 1`] = `
<button class="">
  <div class="b-icon some class name"></div> <span>My Button</span>
  <div class="icon-loading"></div>
</button>
`;

exports[`Admin Button should show the icon if the "iconClass" prop is provided with value 1`] = `
<button class="">
  <div class="b-icon some class name"></div> <span>My Button</span>
  <!---->
</button>
`;

exports[`Admin Button should show the loading icon when "isLoading" is set as "true" 1`] = `
<button class="">
  <!----> <span>My Button</span>
  <div class="icon-loading"></div>
</button>
`;
